src/Agent.Listener/JobDispatcher.cs

```diff
@@ -602,6 +605,34 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, WorkerDisp
                                 var messageType = MessageType.CancelRequest;
                                 if (HostContext.AgentShutdownToken.IsCancellationRequested)
                                 {
+                                    var service = HostContext.GetService<IFeatureFlagProvider>();
+                                    var ffState = await service.GetFeatureFlagAsync(HostContext, "DistributedTask.Agent.FailJobWhenAgentDies", Trace);
+                                    if (ffState.EffectiveState == "On")
+                                    {
+                                        resultOnAbandonOrCancel = TaskResult.Failed;
+                                    }
+                                    try
+                                    {
+                                        var telemetryData = new Dictionary<string, string>
+                                        {
+                                            { "JobId", message.JobId.ToString()},
+                                            { "JobResult", resultOnAbandonOrCancel.ToString() },
+                                        };
+                                        var cmd = new Command("telemetry", "publish")
+                                        {
+                                            Data = JsonConvert.SerializeObject(telemetryData)
+                                        };
+                                        cmd.Properties.Add("area", "PipelinesTasks");
+                                        cmd.Properties.Add("feature", "AgentShutdown");
+
+                                        var telemetryPublisher = HostContext.GetService<IAgenetListenerTelemetryPublisher>();
+
+                                        await telemetryPublisher.PublishEvent(HostContext, cmd);
+                                    }
+                                    catch (Exception ex)
+                                    {
+                                        Trace.Warning($"Unable to publish agent shutdown telemetry data. Exception: {ex}");
+                                    }
```

Comment:
**not a blocker**; feel free to won't fix

eating exceptions from telemetry is okay, but consider removing the exception handler, especially if you tested the telemetry emission well and are assured it won't throw during normal circumstances.  The exception handler may make it harder to detect if there is a bug in the telemetry emission